var/spack/repos/builtin/packages/py-tensorboardx/package.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class PyTensorboardx(PythonPackage):
+    """The purpose of this package is to let researchers use
+       a simple interface to log events within PyTorch (and
+       then show visualization in tensorboard). This package
+       currently supports logging scalar, image, audio,
+       histogram, text, embedding, and the route of back-propagation."""
+
+    homepage = "https://github.com/lanpa/tensorboardX"
+    url      = "https://pypi.io/packages/source/t/tensorboardx/tensorboardX-1.8.tar.gz"
+
+    version('2.0', sha256='835d85db0aef2c6768f07c35e69a74e3dcb122d6afceaf2b8504d7d16c7209a5')
+    version('1.9', sha256='2505d0092e6212f04c4522eea7123e8886c4d0a0b2c406e480fa61ca3c1da7ea')
+    version('1.8', sha256='13fe0abba27f407778a7321937190eedaf12bc8c544d9a4e294fcf0ba177fd76')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-six', type=('build', 'run'))
+    depends_on('py-protobuf@3.8.0:', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-future', type=('build', 'run'))
+    depends_on('py-torchvision', type=('build', 'run'))
+    depends_on('py-torch', type=('build', 'run'))
+    depends_on('py-pillow', type=('build', 'run'))
+    depends_on('py-matplotlib', type=('build', 'run'))
+    depends_on('py-requests', type=('build', 'run'))
+    depends_on('py-onnx', type=('build', 'run'))
+    depends_on('py-packaging', type=('build', 'run'))
+    depends_on('py-boto3', type=('build', 'run'))
+    depends_on('ffmpeg', type=('build', 'run'))
+    depends_on('py-tensorboard', type=('build', 'run'))
```

Comment:
I don't see `future` -> `tensorboard` listed in the `setup.py`. Are you sure these are actually dependencies?

Reply:
I may be going a little overboard here, but see below. I might add chainer/theano later.

```
coltrane (GIT/tensorboardX):pydeps --external tensorboardX
[
    "PIL", 
    "boto3", 
    "builtins", 
    "caffe2", 
    "chainer", 
    "google", 
    "matplotlib", 
    "moviepy", 
    "numpy", 
    "onnx", 
    "six", 
    "theano", 
    "torch", 
    "visdom"
]
['PIL', 'boto3', 'builtins', 'caffe2', 'chainer', 'google', 'matplotlib', 'moviepy', 'numpy', 'onnx', 'six', 'theano', 'torch', 'visdom']

```